src/Provider/AbstractProvider.php

```diff
@@ -323,17 +321,12 @@ protected function fetchProviderData($url)
         try {
             $client = $this->getHttpClient();
 
-            $options = [];
+            $httpResponse = $client->get($url, $this->headers ?: []);
 
-            if ($this->headers) {
-                $options['headers'] = $this->headers;
-            }
-
-            $request = $client->get($url, $options);
-            $response = $request->getBody();
-        } catch (BadResponseException $e) {
+            $response = (string) $httpResponse->getBody();
+        } catch (HttpAdapterException $e) {
             // @codeCoverageIgnoreStart
-            $raw_response = explode("\n", $e->getResponse());
+            $raw_response = explode("\n", (string) $e->getResponse()->getBody());
```

Comment:
As soon as you have merged the 0.x into the 1.0 branch, I can rebase the PR and handle conflicts, if there are any. But this PR is very similar to #223 (which I targeted at 0.x) so I have a good feeling :)